### PR TITLE
Add a numeric format for geographic coordinates

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

File: tests/src/python/test_qgsnumericformat.py [hunk 1]
```diff
@@ -769,6 +770,653 @@ def testFractionFormat(self):
...
+        self.assertEqual(f.formatDouble(-5.44, context), "-005°26.40′")
+
+        f.setShowTrailingZeros(False)
+        f.setShowDirectionalSuffix(True)
+        f.setShowDegreeLeadingZeros(False)
+        self.assertEqual(f.formatDouble(5.44, context), "5°26.4′E")
+        self.assertEqual(f.formatDouble(-5.44, context), "5°26.4′W")
+        self.assertEqual(f.formatDouble(-5.44101, context), "5°26.46′W")
+
+        context.setDecimalSeparator('☕')
```
Comment (nirvn) [inline tests/src/python/test_qgsnumericformat.py:1149]: :laughing:

**Review by nirvn — APPROVED** [review]
Nice!

**Comment by nyalldawson** [comment]
@nirvn I've just added a couple more commits -- do you mind re-looking?

**Comment by nirvn** [comment]
@nyalldawson , looking good.

**Pull request merged**